execution/function/functions.go

```diff
@@ -927,7 +927,7 @@ func dateWrapper(fa FunctionArgs, f func(time.Time) float64) promql.Sample {
 		}
 	}
 	t := time.Unix(int64(fa.Samples[0].F), 0).UTC()
-	lbls, _ := DropMetricName(fa.Labels)
+	lbls, _ := DropMetricName(fa.Labels, labels.NewScratchBuilder(fa.Labels.Len()))
```

Comment:
This also seems like a potential improvement opportunity, but we don't need to do it now since we are not making things worse.

Reply:
lemme see.

Reply:
PTAL :D